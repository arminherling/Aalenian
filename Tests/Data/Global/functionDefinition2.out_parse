FunctionDefinitionStatement: {
    NameExpression: func
    ParametersNode(0): {
    }
    BlockNode(2): {
        AssignmentStatement: {
            Left: {
                NameExpression: a
            }
            Right: {
                NumberLiteral: 0
            }
        }
        ReturnStatement: {
            NameExpression: a
        }
    }
}
