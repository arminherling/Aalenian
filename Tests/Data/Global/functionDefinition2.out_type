TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->(i32)
    Name: func
    Parameters(0): {
    }
    ReturnType: {
        Type: i32
    }
    Body(2): {
        TypedAssignmentStatement: {
            Type: i32
            Left: {
                TypedVariable: {
                    Type: i32
                    Name: a
                }
            }
            Right: {
                I32Value: {
                    Type: i32
                    Value: 0
                }
            }
        }
        TypedReturnStatement: {
            Type: i32
            Expression: {
                TypedVariable: {
                    Type: i32
                    Name: a
                }
            }
        }
    }
}
