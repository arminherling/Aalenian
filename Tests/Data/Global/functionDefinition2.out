FunctionDefinitionStatement: {
    Name: func
    Parameters(0): {
    }
    Block(2): {
        AssignmentStatement: {
            Left: {
                Name: a
            }
            Right: {
                Number: 0
            }
        }
        ReturnStatement: {
            Name: a
        }
    }
}
